### PR TITLE
Improve shop creator with phpstan

### LINKED ISSUE #23: Improve this repository, add tests
This tool is doing amazing things but it can be improved. For example code complexity of class https://github.com/PrestaShop/prestashop-shop-creator/blob/master/src/Generator/EntityGenerator.php is high.

First thing needed to enable refactoring of this repository would be to add automated tests, and run them in a CI using GitHub Actions.

When this is done we will be able to use CI to detect whether our refactoring is breaking an existing feature / behavior or not.

I suppose that we can use this opportunity to add the standard CI helpers: phpstan, php-cs-fixer... see for example https://github.com/PrestaShop/ps_linklist/blob/dev/.github/workflows/php.yml

### PR DESCRIPTION
**Pull request by mflasquin**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Improve shop creator with phpstan
| Type?             | improvement
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #23 
| How to test?      | see https://github.com/mflasquin/prestashop-shop-creator/pull/4
| Possible impacts? | Please indicate what parts of the software we need to check to make sure everything is alright.

### CODE REVIEW

File: .github/workflows/php.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+          restore-keys: ${{ runner.os }}-composer-
+
+      - name: Composer Install
+        run: |
+          composer install --ansi --prefer-dist --no-interaction --no-progress
+          rm composer.lock
+          composer config platform.php ${{ matrix.php }}
+
+      - name: Run phpstan
+        run: ./vendor/bin/phpstan analyse
```
Comment (nicosomb) [inline .github/workflows/php.yml:46]: new line needed 😉

**Review by nicosomb — DISMISSED** [review]

File: .github/workflows/php.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    name: PHP Static Analysis
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        php: [ '7.2', '7.3', '7.4', '8.0', '8.1' ]
+      fail-fast: false
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.9.0
+        with:
```
Comment (lartist) [inline .github/workflows/php.yml:17]:
```suggestion
        with:
          all_but_latest: true
```

**Review by matks — CHANGES_REQUESTED** [review]
PR is ok but needs to be rebased to fix git conflicts

Please rebase and ping me and then I will approve ;)

**Comment by mflasquin** [comment]
> PR is ok but needs to be rebased to fix git conflicts
> 
> Please rebase and ping me and then I will approve ;)

Hi @matks, done :smile:

**Review by matks — APPROVED** [review]

**Comment by matks** [comment]
Thank you @mflasquin

**Pull request merged**